3_javascript/javascript1/week1/javascript.js

```diff
@@ -0,0 +1,104 @@
+ //hello... Hello... This is my first experience in Javascript :)
+
+console.log("******    Age-ify (A future age calculator)    ******");
+let myYearOfBirth = 1991;
+let yearFuture = 2027;
+let age = yearFuture - myYearOfBirth;
+
+console.log("I\'ll be " + age + " years old in " + yearFuture + ".");
+console.log("\n");
+
+
+
+console.log("******    Goodboy-Oldboy (A dog age calculator)    ******");
+let dogYearOfBirth = 2017;
+let dogYearFuture = 2045;
+// We have a conditional statement, so a "BOOLEAN variable" can represent (shouldShowResultInDogYears).
+let shouldShowResultInDogYears = true;
+
+ // According to the information in the txt:(1 human year = 7 dog year)
+let dogAge = (dogYearFuture - dogYearOfBirth)*7;
+let humanAge = dogYearFuture - dogYearOfBirth;
+
+if (shouldShowResultInDogYears) {
+    console.log("His dog will be " + dogAge + " dog years old in " + dogYearFuture + ".");
+}
+
+else {
+ console.log("His dog will be " + humanAge + " human years old in " + dogYearFuture + ".");
+}
+console.log("/* His dog will be 28 human years old in 2045. */");
+console.log("\n");
+
+
+console.log("******    Housey pricey (A house price estimator)    ******");
+//Peter:
+var petersPreferredHouseWide = a;
+var petersPreferredHouseDeep = b;
+var petersPreferredHouseHigh = c;
+
+let petersGardenSizeInM2 = 100; //m2
+let petersPreferredHouseCosts = 2500000; //$
+
+var a = 8; //m
+var b = 10; //m
+var c = 10; //m
+
+var volumeInMeters = a * b * c;
+const petersHousePrice = volumeInMeters * 2.5 * 1000 + petersGardenSizeInM2 * 300;
+const petersPriceDifference = petersPreferredHouseCosts - petersHousePrice;
+//We can consider the currency unit as the US Doller.
+console.log("Peter is paying " + petersPriceDifference + " more than real price.");
+
+//Julia:
+var juliasPreferredHouseWide = x;
+var juliasPreferredHouseDeep = y;
+var juliasPreferredHouseHigh = z;
+
+let juliasGardenSizeInM2 = 70; //m2
+let juliasPreferredHouseCosts = 1000000; //$
+
+var x = 5; //m
+var y = 11; //m
+var z = 8; //m
+
+var volumeInMeters = x * y * z;
+const juliasHousePrice = volumeInMeters * 2.5 * 1000 + juliasGardenSizeInM2 * 300;
+const juliasPriceDifference = juliasPreferredHouseCosts - juliasHousePrice;
+//We can consider the currency unit as the US Doller.
+console.log("Julia is paying " + juliasPriceDifference + " less than real price.");
+console.log("\n");
+
+
+console.log("******    Ez Namey (Startup name generator) Optional    ******");
+const firstWords =  [
+    "Golden",
+    "Amazing",
+    "Brilliant",
+    "Cute",
+    "Inteligent",
+    "Smart",
+    "Silver",
+    "Shine",
+    "Alpha",
+    "Fast"
+];
+
+const secondWords = [
+    "Magic",
+    "Mind",
+    "Ocion",
+    "Atlas",
+    "Persian",
+    "Maad",
+    "Journey",
+    "Team",
+    "Engeener",
+    "Code"
+];
+
+const randomNumber = Math.floor(Math.random() * 10);
+console.log(randomNumber);
+const startupName = firstWords[randomNumber] + " " + secondWords[randomNumber];
```

Comment:
Good Job @helena69hme! The only one caveat is counting the characters in ﻿`startupName` with `﻿startupName.length`, it also includes the space (﻿+ " " +). Since we do not want to include this in our value, we have two options:

1. Declare the names separately and then add them together.
```
const firstWord = firstWords[randomNumber] 
const secondWord = secondWords[randomNumber];
const numOfCharacters = firstWord.length + secondWord.length
console.log("The startup: " + startupName + " contains " + numOfCharacters + " characters");
```
2. simple substruct 1 from the value like this
`console.log("The startup: " + startupName + " contains " + startupName.length - 1 + " characters");`

Reply:
Dear @BrantalikP
I followed up your advice step by step...Done now although it was a little challenging. I'm a bit confused.
Let me ask you why by using the second substruct which you offered, this message appeares on the js console:
"NaN characteres"

![js length](https://github.com/HackYourFuture-CPH/class27-homework/assets/142342651/e5f621b9-6eb2-47cf-bb1a-f91a10f8bae8)
![jv length concole](https://github.com/HackYourFuture-CPH/class27-homework/assets/142342651/48e058b4-3603-4c38-83ab-447b12e86337)

Reply:
Hey @helena69hme 

Sorry this is my mistake, I wrote the code here just as plain example and did not realize the mistake I made.

You should subtract the number first in separate variable, then use it inside of the string. 

like this:

```
 const count = startupName.length - 1;
 console.log("The startup: " + startupName + " contains " + count + " characters");
```

this should work :)

Reply:
oh!  I see :)
you are not who should be sorry... I must persist in finding the answer, with more practice, deeper exploration, and trying different paths. I am infinitely grateful for your explanation and attention.